sprint.rb

```diff
@@ -20,27 +21,22 @@ def self.recent_sprints(count)
     sprints(as_of: nil).slice_after { |s| s.date >= Date.today }.first.last(count)
   end
 
-  def self.sprints(as_of: Date.today)
-    require "active_support/core_ext/numeric/time"
-    # The first date when we started doing this cadence
-    number = 76
-    date   = Date.parse("Jan 1, 2018")
-    range  = Date.parse("Dec 12, 2018")..date
+  def self.last_completed
+    Sprint.recent_sprints(2).first
+  end
 
-    as_of ||= date
+  def self.create_by_sprint_number(sprint_number)
+    SprintBoundaryIterator.new.each do |number, range|
+      return new(number, range) if number == sprint_number
+    end
```

Comment:
This can be

```ruby
found = SprintBoundaryIterator.new.detect { |number, _range| number == sprint_number }
found ? new(*found) : nil
```

Reply:
However, if it's never found, this will run forever as the iterator is an infinite iterator.

Reply:
The SprintBoundaryIterator is not an infinite iterator.  The SprintBoundaryIterator#each method only emits [number, range] pairs in the past.